Fix missing HTML body content for IMAP feeds

deafmute1 opened the pull request:
Addresses #36 .

The existing implementation uses the `<description>` field for a plain text version of the content, `<content:encoded>` field for the full html email body. 

The rss library  does not appear to have any support for the `<content>` field at all. Output XML does not include it, despite it being generated fine before being passed to the library. 

There is no mention of it in their `itemOption` documentation, and the documentation explicitly says to use the `description` field: 
> description string Content for the item. Can contain html but link and image urls must be absolute path including hostname.  (https://github.com/dylang/node-rss?tab=readme-ov-file#itemoptions)

The commit removes the separate content fields and merges them into one for the `<description>` tag.

Additionally, there is a small fix I noticed when testing on the line above to prevent `&nbsp` being "escaped" into `&amp;nbsp`

TBosak commented:
> It is actually possible to add a content:encoded field like this apparently:
> 
> ```
> feed.item({
>     title:  'Item Title ',
>     description: 'The description',
>     custom_elements: [
>       {
>         "content:encoded":
>             {
>                 _cdata: "This is the long content. <b>This & That</b>"
>             }
>       }
>     ]
> });
> ```
> 
> this is probably a better fix.

Want to go ahead and implement it, and I'll review & test on my end before approving?

deafmute1 commented:
@TBosak  I've just implemented it.

In the course of doing so I noticed that the RSS Models didn't match the API of of `rss.feedOptions` (i.e. customElement instead of custom_elements). I ended up just bringing all models in the entire file in line with `rss`. There are incorrectly named/non-existent fields in `rss-feed.model.ts` that I have not fixed, that may be breaking other parts of feed generation for non-IMAP feeds.

Additionally, I just removed the `email:to` and `email:cc` fields from custom_elements. Previously, they did nothing as the field was incorrectly named. After updating the field, they resulted in the output XML being invalid because the namespace `email` is undefined. I could not find a spec document for `email` on google, so I just removed it entirely.

TBosak commented:
> @TBosak I've just implemented it.
> 
> In the course of doing so I noticed that the RSS Models didn't match `rss.feedOptions` & `rss.itemOptions` (i.e. customElement in model but its custom_elements in `rss.itemOptions`). I ended up just bringing all models in the entire file in line with `rss`. There are incorrectly named/non-existent fields in `rss-feed.model.ts` that I have not fixed, that may be breaking other parts of feed generation for non-IMAP feeds.
> 
> Additionally, I just removed the `email:to` and `email:cc` fields from custom_elements. Previously, they did nothing as `custom_elements` was incorrectly named. After updating the field, they resulted in the output XML being invalid because the namespace `email` is undefined. I could not find a spec document for `email` on google, so I just removed it entirely.

I went ahead and just switched to _feed_ in place of _node-rss_, as it seems more recent and is maintained. I aligned everything with the new types. "Content" is included in the feed items for _feed_, but some image metadata that I was handling before (I'm not sure if _node-rss_ was even outputting this originally) is no longer being handled. Some other metadata has been moved to extensions (custom elements) as well.
